memiavl/tree.go

```diff
@@ -68,10 +81,12 @@ func (t *Tree) ApplyChangeSet(changeSet *iavl.ChangeSet, updateHash bool) ([]byt
 }
 
 func (t *Tree) set(key, value []byte) {
+	t.bwal.addChange(preparePair(key, value))
```

Comment:
now you can add the whole change set in `ApplyChangeSet` at once?

Reply:
yeah, I refactored the code today to use the changeset PR's logic. So basically wal works right now like this: it stores a pair in memory when a `set/remove` is called, and writes to the underlying wal every block's changeset when a `saveVersion` is called.

Right now I am working on `ReplayWAL` function that will basically take the tree at version X, and replay all the changesets stored in a wal until the desired height.

If you have any suggestions on the design, please, lmk!

Reply:
I think you can write the change set passed in ApplyChangeSet method directly into WAL, since that's the only entry method now, set/remove won't be called directly by user.

Reply:
yeah, makes sense, will do that